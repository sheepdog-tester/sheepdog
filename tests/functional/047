#!/bin/bash

# Test journal file replay

. ./common

_start_sheep 0 "-j size=64M"

_wait_for_sheep 1

_cluster_format -c 1

$DOG vdi create test 4G

# write something to vdi 'test'
echo "hello" | $DOG vdi write test 0 512
echo "sheepdog" | $DOG vdi write test 4M 512
# corrupt the vdi...
_kill_sheep 0
if $MD; then
	rm $STORE/0/d*/807c2b2500000000
	rm $STORE/0/d*/007c2b25*
else
	rm $STORE/0/obj/807c2b2500000000
	rm $STORE/0/obj/007c2b25*
fi

# do the journal replay
_start_sheep 0 "-j size=64M"
_wait_for_sheep 1

sleep 3

_vdi_list
$DOG vdi read test 0 512 | tr -d [:cntrl:]
echo -n ' '
$DOG vdi read test 4M 512 | tr -d [:cntrl:]
echo ''
