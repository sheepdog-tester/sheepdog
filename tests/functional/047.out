QA output created by 047
using backend plain store
  Name        Id    Size    Used  Shared    Creation time   VDI id  Copies  Tag   Block Size Shift
  test         0  4.0 GB  8.0 MB  0.0 MB DATE   7c2b25      1                22
hello sheepdog
